Source/Core/Core/HW/WiimoteEmu/Encryption.cpp

```diff
@@ -491,15 +565,49 @@ EncryptionKey KeyGen1stParty::GenerateTables(const EncryptionKey::RandData& rand
                                              const EncryptionKey::KeyData& key, u32 idx) const
 {
   EncryptionKey result;
-  result.GenerateTables(rand, key, idx + 1, 0);
+  result.GenerateTables(rand, key, sboxes_1st_party[idx], sboxes_1st_party[idx + 1]);
   return result;
 }
 
 EncryptionKey KeyGen3rdParty::GenerateTables(const EncryptionKey::RandData& rand,
                                              const EncryptionKey::KeyData& key, u32 idx) const
 {
   EncryptionKey result;
-  result.GenerateTables(rand, key, idx + 10, 0x30);
+  result.GenerateTables(rand, key, sboxes_3rd_party[idx], sboxes_3rd_party[idx + 1]);
+  return result;
+}
+
+EncryptionKey KeyGen1stParty::GenerateFallbackTables(const EncryptionKey::RandData& rand,
+                                                     const EncryptionKey::KeyData& key) const
+{
+  // Reverse engineering did not produce exact tables, only the XOR effect of each.
+  // This is the XOR difference between the tables.
+  // This also works with 0x97.
```

Comment:
I'm not the target of this kind of information but what does this mean?  That using `0x97` also produces the same result logically?  Why did you choose `0x17` over `0x97`?

Reply:
Yes. It's the same result logically.
The encryption function is: `enc = (val - ft) ^ sb`
`ft` and `sb` are produced from an XOR of values from each of the reverse engineered tables.
Subtracting, Adding, or XORing 0x80 to a u8 all produce the same result so if the table XOR difference is 0x80 more the end result is the same.
I chose 0x17 because it's a lower number. :P
Some more info here: https://wiibrew.org/wiki/Wiimote/Extension_Controllers